lib/puppet/indirector/s3.rb

```diff
@@ -0,0 +1,47 @@
+require 'puppet/file_serving/terminus_helper'
+require 'puppet/indirector/terminus'
+require 'aws-sdk-core' if Puppet.features.aws?
```

Comment:
Is there a better way to restrict the S3 file source than placing it here?

Looking at some other parts of the puppet code them seem to use a confine statement?